### ISSUE TITLE
gie parses input incorrectly in some cases

### ISSUE THREAD

**Issue by kbevers** [OSGeo/PROJ#623]
As demonstrated below. I did a quick debug session and determined that the expect coordinates are not properly read. If I change the "0" to "0." or "0.0" the test passes. Similar to the recent issue with cct, although the fix to that should apply here as well.

```
$ cat dev/proj4/test/fail.gie
BEGIN

--------------------------------------------------
OPERATION +proj=pipeline +step +init=epsg:4283 +inv +step +init=epsg:3577
TOLERANCE 0.05m
--------------------------------------------------
ACCEPT    132 -27
EXPECT    0 -2926820.89

END                                                                                                                                                            
$ gie dev/proj4/test/fail.gie
-------------------------------------------------------------------------------
Reading file 'dev/proj4/test/fail.gie'
-------------------------------------------------------------------------------
+proj=pipeline +step +init=epsg:4283 +inv +step +init=epsg:3577
-------------------------------------------------------------------------------
     FAILURE in fail.gie(8):
     expected: 0 -2926820.89
     got:      0.000000000   -2926820.886839157   0.000000000   0.000000000
     deviation:  9999.999 mm,  expected:  50.000 mm
-------------------------------------------------------------------------------
total:  0 tests succeeded,   1 tests FAILED!
-------------------------------------------------------------------------------
```

**Comment by busstoptaktik** [comment]
Just to be sure: you are building against the version from merged PR #614?

I will add tests for the  "0  " and "0_"  cases ("0" works)

**Comment by kbevers** [comment]
Yes. Built with 05eed057d1
